cmake/AddFdbTest.cmake

```diff
@@ -380,8 +380,127 @@ function(create_valgrind_correctness_package)
   endif()
 endfunction()
 
+function(prepare_binding_test_files build_directory target_name target_dependency)
+  add_custom_target(${target_name} DEPENDS ${target_dependency})
+  add_custom_command(
+    TARGET ${target_name}
+    COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:fdb_flow_tester> ${build_directory}/tests/flow/bin/fdb_flow_tester
+    COMMENT "Copy Flow tester for bindingtester")
+
+  set(generated_binding_files python/fdb/fdboptions.py)
+  if(WITH_JAVA_BINDING)
+    if(NOT FDB_RELEASE)
+      set(not_fdb_release_string "-SNAPSHOT")
+    else()
+      set(not_fdb_release_string "")
+    endif()
+    add_custom_command(
+      TARGET ${target_name}
+      COMMAND ${CMAKE_COMMAND} -E copy
+        ${CMAKE_BINARY_DIR}/packages/fdb-java-${FDB_VERSION}${not_fdb_release_string}.jar
+        ${build_directory}/tests/java/foundationdb-client.jar
+      COMMENT "Copy Java bindings for bindingtester")
+    add_dependencies(${target_name} fat-jar)
+    add_dependencies(${target_name} foundationdb-tests)
+    set(generated_binding_files ${generated_binding_files} java/foundationdb-tests.jar)
+  endif()
+
+  if(WITH_GO_BINDING)
+    add_dependencies(${target_name} fdb_go_tester fdb_go)
+    add_custom_command(
+      TARGET ${target_name}
+      COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_BINARY_DIR}/bindings/go/bin/_stacktester ${build_directory}/tests/go/build/bin/_stacktester
+      COMMAND ${CMAKE_COMMAND} -E make_directory ${build_directory}/tests/go/src/fdb/
+      COMMAND ${CMAKE_COMMAND} -E copy
+        ${CMAKE_BINARY_DIR}/bindings/go/src/github.com/apple/foundationdb/bindings/go/src/fdb/generated.go # SRC
+        ${build_directory}/tests/go/src/fdb/ # DEST
+      COMMENT "Copy generated.go for bindingtester")
+  endif()
+
+  foreach(generated IN LISTS generated_binding_files)
+    add_custom_command(
+      TARGET ${target_name}
+      COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_BINARY_DIR}/bindings/${generated} ${build_directory}/tests/${generated}
+      COMMENT "Copy ${generated} to bindingtester")
+  endforeach()
+endfunction(prepare_binding_test_files)
+
+function(package_bindingtester2)
+  if (WIN32 OR OPEN_FOR_IDE)
+    message(WARNING "Binding tester is not built (WIN32/OPEN_FOR_IDE)")
+    return()
+  endif()
+
+  set(fdbcName "libfdb_c.so")
+  if (APPLE)
+    set(fdbcName "libfdb_c.dylib")
+  endif ()
+
+  set(touch_file ${CMAKE_BINARY_DIR}/bindingtester2.touch)
```

Comment:
why do we need this file?

Reply:
We need to set up a directory to store temporary files to be packed into the bindingtester tar file before copying the actual contents. The directory has to be bound to a target or a file in order to ensure the ordering of the execution. Since the previous code, package_bindingtester, is using a file, here I used a file so I can reuse the old code.